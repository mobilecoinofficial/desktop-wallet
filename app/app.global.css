/*
 * @NOTE: Prepend a `~` to css file paths that are in your node_modules
 *        See https://github.com/webpack-contrib/sass-loader#imports
 */
@import '~@fortawesome/fontawesome-free/css/all.css';

body {
  position: relative;
  color: white;
  height: 100vh;
  background-color: black;
  font-family: 'Lucida Console', Monaco, monospace;
  overflow: auto;
  letter-spacing: 4px;
}

.MuiTabs-flexContainer {
  justify-content: space-between;
}
