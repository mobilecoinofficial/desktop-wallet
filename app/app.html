<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta
      http-equiv="Content-Security-Policy"
      content="default-src 'none';
                   img-src 'self';
                   connect-src ws://localhost:1212/
                               http://localhost:1212/
                               http://localhost:9090;
                   script-src 'self'
                               http://localhost:1212/
                               http://localhost:9090
                              'unsafe-eval'
                              'sha256-1IUHPNXweUJ6kDru06FXRLzQKpg7vV9HfCeHGSX1D7A='
                              'sha256-f7gMtshsVI7mmPkQlBpl9zRvFqFSagSGF35HpPJWVN0=';
                   style-src 'self'
                             http://localhost:1212/
                             http://localhost:9090
                             'unsafe-inline';
                   font-src 'self'
                            http://localhost:1212/
                            http://localhost:9090;
                   object-src 'none';"
    />
    <title>MobileCoin Desktop Wallet</title>
    <script>
      (() => {
        if (
          typeof process !== 'object' ||
          (typeof process === 'object' && !process.env.START_HOT)
        ) {
          const link = document.createElement('link');
          link.rel = 'stylesheet';
          link.href = './dist/style.css';
          // HACK: Writing the script path should be done with webpack
          document.getElementsByTagName('head')[0].appendChild(link);
        }
      })();
    </script>
  </head>
  <body>
    <div id="root"></div>
    <script>
      if (typeof process === 'object') {
        const scripts = [];

        if (process.env.NODE_ENV === 'development') {
          // Dynamically insert the DLL script in development env in the
          // renderer process
          scripts.push('../dll/renderer.dev.dll.js');
        }
        if (process.env.START_HOT) {
          // Dynamically insert the bundled app script in the renderer process
          const port = process.env.PORT || 1212;
          scripts.push(`http://localhost:${port}/dist/renderer.dev.js`);
        } else {
          scripts.push('./dist/renderer.prod.js');
        }

        if (scripts.length) {
          document.write(
            scripts.map((script) => `<script defer src="${script}"><\/script>`).join('')
          );
        }
      }
    </script>
  </body>
</html>
