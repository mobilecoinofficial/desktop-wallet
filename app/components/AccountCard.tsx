import React, { useState } from 'react';
import type { FC } from 'react';

import {
  Box,
  Card,
  CardContent,
  Container,
  IconButton,
  Tooltip,
  Typography,
  makeStyles,
} from '@material-ui/core';
import { clipboard } from 'electron';
import { useSnackbar } from 'notistack';
import { useTranslation } from 'react-i18next';

import type { Theme } from '../theme';
import LongCode from './LongCode';
import QRMob from './QRMob';
import ShortCode from './ShortCode';
import { CodeTextIcon, LogoIcon, QRCodeIcon } from './icons';

interface AccountCardProps {
  account: { b58Code: string; name?: string };
  isGift?: boolean;
}

const useStyles = makeStyles((theme: Theme) => ({
  code: {
    cursor: 'pointer',
    display: 'flex',
    flexDirection: 'column',
    fontSize: 22,
    letterSpacing: '.65rem',
    marginRight: '-.65rem',
    padding: theme.spacing(1),
  },
  container: {
    padding: '0',
  },
  corners: {
    alignItems: 'center',
    display: 'flex',
    justifyContent: 'space-between',
    width: '100%',
  },
  root: {
    padding: '0 10px',
  },
}));

const AccountCard: FC<AccountCardProps> = ({ account, isGift, ...rest }: AccountCardProps) => {
  const [isQRCode, setIsQRCode] = useState(false);
  const classes = useStyles();
  const { enqueueSnackbar = () => {} } = useSnackbar() || {};
  const { t } = useTranslation('AccountCard');

  const { b58Code, name } = account;

  const mobUrl = `mob:///b58/${b58Code}`;

  const handleCodeClick = (code: string) => () => {
    clipboard.writeText(code);
    enqueueSnackbar(isGift ? t('clipboardGift') : t('clipboardAddress'), {
      variant: 'success',
    });
  };

  const handleToggleClick = () => {
    setIsQRCode(!isQRCode);
  };

  let headerString = '';
  if (isQRCode) {
    headerString = isGift ? t('giftQR') : t('accountQR');
  } else {
    headerString = isGift ? t('giftHeader') : t('accountHeader');
  }

  return (
    <Container className={classes.container} fixed maxWidth="sm">
      <Card data-testid="account-card" className={classes.root} {...rest}>
        <CardContent>
          <Box
            display="flex"
            alignItems="center"
            flexDirection="column"
            textAlign="center"
            justifyContent="space-between"
            minHeight={400}
          >
            <Box className={classes.corners}>
              <LogoIcon />
              <Tooltip
                title={isQRCode ? t('accountTooltip') : t('mobUrlTooltip')}
                placement="right"
                arrow
              >
                <IconButton
                  data-testid="account-card-toggle"
                  color="inherit"
                  edge="start"
                  onClick={handleToggleClick}
                >
                  {isQRCode ? <CodeTextIcon /> : <QRCodeIcon />}
                </IconButton>
              </Tooltip>
            </Box>
            <Typography color="textSecondary">{headerString}</Typography>
            <Typography
              data-testid="account-card-center"
              color="textPrimary"
              gutterBottom
              variant="h3"
            >
              {isQRCode ? (
                <QRMob data-testid="account-card-qr-code" size={280} value={mobUrl} />
              ) : (
                <Tooltip title={t('copyTooltip')} placement="right" arrow>
                  <Box data-testid="account-card-tooltip" onClick={handleCodeClick(b58Code)}>
                    <LongCode
                      data-testid="account-card-long-code"
                      codeClass={classes.code}
                      code={b58Code}
                    />
                  </Box>
                </Tooltip>
              )}
            </Typography>
            <Box className={classes.corners}>
              <Typography data-testid="account-card-name" color="textPrimary" variant="h4">
                {name || t('unnamed')}
              </Typography>
              <Typography data-testid="account-card-short-code" color="textSecondary" variant="h4">
                <ShortCode code={b58Code} />
              </Typography>
            </Box>
          </Box>
        </CardContent>
      </Card>
    </Container>
  );
};

AccountCard.defaultProps = {
  isGift: false,
};

export default AccountCard;
