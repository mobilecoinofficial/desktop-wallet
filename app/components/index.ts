export { default as AccountCard } from './AccountCard';
export { default as GlobalStyles } from './GlobalStyles';
export { default as LoadingScreen } from './LoadingScreen';
export { default as MOBNumberFormat } from './MOBNumberFormat';
export { default as QRMob } from './QRMob';
export { default as QRScanner } from './QRScanner';
export { default as SavedPasswordsModal } from './SavedPasswordsModal';
export { default as ShortCode } from './ShortCode';
export { default as SplashScreen } from './SplashScreen';
export { default as StarCheckbox } from './StarCheckbox';
export { default as SubmitButton } from './SubmitButton';
export { default as TabPanel } from './TabPanel';
export { default as TermsOfUse } from './TermsOfUse.component';
export { default as TermsOfUseDialog } from './TermsOfUse.dialog';
export { default as TransactionInfoLabel } from './TransactionInfoLabel';
