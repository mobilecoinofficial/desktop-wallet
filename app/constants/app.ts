export const APP_VERSION = process.env.npm_package_version;

export const MIN_WINDOW_HEIGHT = 900;
export const MIN_WINDOW_WIDTH = 720;

export const HISTORY_PAGE_SIZE = 5; // for usage in Transaction History

export const INITIAL_WINDOW_HEIGHT = 880;

export const TIME_FOR_INACTIVITY = 300000; // after this time, will be considered inactive: 5'
export const TIME_FOR_REACTION = 30000; // extra time to avoid closing: 30"

export const RANDOM_COLORS = ['#8B35E0', '#1F639A', '#EAA520', '#15A389', '#8D969D', '#D82E26'];

export const randomColor = (): string =>
  RANDOM_COLORS[Math.floor(RANDOM_COLORS.length * Math.random())];
