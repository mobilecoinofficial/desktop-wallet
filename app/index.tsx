/* eslint-disable @typescript-eslint/no-var-requires */
import React, { Fragment } from 'react';

import { ipcRenderer } from 'electron';
import { render } from 'react-dom';
import { AppContainer as ReactHotAppContainer } from 'react-hot-loader';

import i18n from './i18n';

import './app.global.css';

const AppContainer = process.env.PLAIN_HMR ? Fragment : ReactHotAppContainer;

ipcRenderer.sendSync('get-initial-translations');

ipcRenderer.on('language-changed', (_, message) => {
  if (!i18n.hasResourceBundle(message.language, message.namespace)) {
    i18n.addResourceBundle(message.language, message.namespace, message.resource);
  }

  i18n.changeLanguage(message.language);
});

document.addEventListener('DOMContentLoaded', () => {
  // eslint-disable-next-line global-require
  const App = require('./App.tsx').default;

  render(
    <AppContainer>
      <App />
    </AppContainer>,
    document.getElementById('root')
  );
});
