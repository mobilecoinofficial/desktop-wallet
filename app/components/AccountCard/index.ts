import { AccountCard } from './AccountCard.view';

export default AccountCard;

export { AccountCard };
