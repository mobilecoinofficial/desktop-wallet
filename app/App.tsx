import React, { useEffect, useState } from 'react';
import type { FC } from 'react';

import { ThemeProvider } from '@material-ui/core';
import { ipcRenderer } from 'electron';
import { SnackbarProvider } from 'notistack';
import { hot } from 'react-hot-loader/root';
import { MemoryRouter } from 'react-router-dom';

import { GlobalStyles } from './components/GlobalStyles';
import { MOBILE_COIN_DARK, MOBILE_COIN_LIGHT } from './constants/themes';
import { FullServiceProvider } from './contexts/FullServiceContext';
import { renderRoutes } from './routes';
import { setTheme } from './theme';

const App: FC = () => {
  const [theme, setThemeReact] = useState(
    setTheme({
      responsiveFontSizes: true,
      theme: ipcRenderer.sendSync('get-theme') === 'light' ? MOBILE_COIN_LIGHT : MOBILE_COIN_DARK,
    })
  );

  useEffect(() => {
    ipcRenderer.on('set-theme-light', () => {
      setThemeReact(
        setTheme({
          responsiveFontSizes: true,
          theme: MOBILE_COIN_LIGHT,
        })
      );
    });
    ipcRenderer.on('set-theme-dark', () => {
      setThemeReact(
        setTheme({
          responsiveFontSizes: true,
          theme: MOBILE_COIN_DARK,
        })
      );
    });

    return () => {
      ipcRenderer.removeAllListeners('set-theme-light');
      ipcRenderer.removeAllListeners('set-theme-dark');
    };
  });

  return (
    <MemoryRouter>
      <ThemeProvider theme={theme}>
        <SnackbarProvider dense maxSnack={5}>
          <FullServiceProvider>
            <GlobalStyles />
            {renderRoutes()}
          </FullServiceProvider>
        </SnackbarProvider>
      </ThemeProvider>
    </MemoryRouter>
  );
};

export default hot(App);
