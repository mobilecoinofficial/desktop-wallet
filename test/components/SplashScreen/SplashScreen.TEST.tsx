import React from 'react';

import { SplashScreen } from '../../../app/components';
import renderSnapshot from '../../renderSnapshot';

describe('SplashScreen', () => {
  test('renders to screen when called', () => {
    const { asFragment } = renderSnapshot(<SplashScreen />);
    expect(asFragment()).toMatchSnapshot();
  });
});
