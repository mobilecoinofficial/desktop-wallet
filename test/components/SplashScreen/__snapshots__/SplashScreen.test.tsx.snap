// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreen renders to screen when called 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root makeStyles-root"
    data-testid="SplashScreen"
  >
    <div
      class="MuiBox-root MuiBox-root"
    >
      <div
        class="MuiBox-root MuiBox-root"
      >
        <svg
          color="#A6AAB4"
          fill="none"
          height="105"
          viewBox="0 0 141 35"
          width="423"
        >
          <path
            d="M44.887 18.642l-4.397-6.15V23.74h-2.245V7.837h1.38l5.59 7.858 5.308-7.858h1.403V23.74h-2.268V12.702l-3.953 5.823-.818.117zM64.929 21.191a5.058 5.058 0 01-2 2.105c-.865.499-1.875.749-3.028.749-1.138 0-2.133-.25-2.982-.749-.85-.499-1.5-1.2-1.953-2.105-.452-.904-.678-1.949-.678-3.133 0-1.17.234-2.206.702-3.11a5.083 5.083 0 011.987-2.106c.858-.499 1.863-.748 3.017-.748 1.138 0 2.132.25 2.982.748.85.5 1.504 1.2 1.965 2.105.46.905.69 1.95.69 3.134 0 1.169-.235 2.206-.702 3.11zm-2.538-.21c.616-.764.932-1.723.947-2.877 0-1.216-.296-2.202-.888-2.958-.593-.756-1.412-1.135-2.456-1.135-1.029 0-1.851.383-2.467 1.146-.616.765-.932 1.723-.947 2.877 0 1.216.296 2.202.888 2.958.593.757 1.411 1.134 2.456 1.134 1.029 0 1.851-.381 2.467-1.146zM75.144 12.842c.85.5 1.5 1.205 1.953 2.117.452.912.678 1.976.678 3.192 0 1.17-.234 2.199-.702 3.087a5.046 5.046 0 01-1.988 2.07c-.857.491-1.847.737-2.97.737-.39 0-.846-.082-1.368-.246a5.39 5.39 0 01-1.485-.736l-.445.678h-1.262V7.488L69.8 7.23v5.566c.42-.218.837-.39 1.251-.514.412-.125.775-.188 1.087-.188 1.154 0 2.155.25 3.005.748zm-.55 8.174c.592-.74.889-1.688.889-2.842 0-1.247-.293-2.252-.877-3.016-.585-.764-1.408-1.146-2.468-1.146-.92 0-1.699.272-2.338.818v6.525c.655.515 1.434.772 2.338.772 1.045 0 1.863-.37 2.456-1.111zM80.61 10.364c-.39 0-.718-.137-.983-.41a1.371 1.371 0 01-.397-.994c0-.39.132-.72.397-.994.265-.272.593-.409.982-.409.39 0 .725.137 1.006.41.28.272.42.604.42.993 0 .39-.14.722-.42.994-.281.273-.616.41-1.006.41zm-1.1 13.376V12.398h2.269V23.74H79.51zM84.335 23.74V7.487l2.245-.258V23.74h-2.245zM98.927 18.572h-8.044c.093 1.108.448 1.98 1.064 2.62.615.639 1.531.958 2.747.958.592 0 1.11-.074 1.555-.222.445-.148 1.064-.41 1.86-.784l.702 1.638c-1.373.826-2.745 1.239-4.116 1.239-1.98 0-3.5-.538-4.56-1.614-1.061-1.075-1.591-2.525-1.591-4.35 0-1.169.218-2.205.655-3.11.436-.904 1.064-1.606 1.883-2.105.818-.499 1.78-.748 2.888-.748 1.637 0 2.872.55 3.707 1.649.834 1.099 1.25 2.506 1.25 4.22v.609zm-7.998-1.52h5.823c-.296-2.026-1.247-3.04-2.853-3.04-.858 0-1.54.269-2.047.806-.506.539-.814 1.283-.923 2.234zM112.023 8.47l-.701 1.613c-.702-.25-1.271-.421-1.708-.515-.436-.093-.998-.14-1.684-.14-.967 0-1.844.269-2.63.807-.788.538-1.408 1.297-1.86 2.28-.452.982-.678 2.097-.678 3.344 0 1.154.198 2.21.596 3.169.398.959.975 1.719 1.731 2.28.756.561 1.641.842 2.654.842 1.31 0 2.471-.234 3.485-.702l.865 1.684c-.515.265-1.185.487-2.012.666-.826.18-1.574.27-2.244.27-1.575 0-2.92-.363-4.034-1.088a6.84 6.84 0 01-2.526-2.947c-.569-1.24-.853-2.623-.853-4.15 0-1.591.307-3.021.923-4.292.616-1.27 1.497-2.268 2.643-2.994 1.146-.725 2.475-1.087 3.987-1.087.562 0 1.247.082 2.058.246.811.164 1.474.402 1.988.713zM123.108 21.191a5.069 5.069 0 01-2 2.105c-.865.499-1.875.749-3.029.749-1.138 0-2.132-.25-2.982-.749a4.941 4.941 0 01-1.952-2.105c-.453-.904-.679-1.949-.679-3.133 0-1.17.234-2.206.702-3.11a5.076 5.076 0 011.988-2.106c.857-.499 1.862-.748 3.016-.748 1.139 0 2.132.25 2.982.748.85.5 1.505 1.2 1.965 2.105.459.905.69 1.95.69 3.134 0 1.169-.234 2.206-.701 3.11zm-2.538-.21c.616-.764.932-1.723.947-2.877 0-1.216-.296-2.202-.888-2.958-.593-.756-1.411-1.135-2.456-1.135-1.028 0-1.851.383-2.467 1.146-.616.765-.932 1.723-.947 2.877 0 1.216.296 2.202.889 2.958.592.757 1.41 1.134 2.455 1.134 1.029 0 1.851-.381 2.467-1.146zM126.644 10.364c-.389 0-.717-.137-.982-.41a1.372 1.372 0 01-.397-.994c0-.39.132-.72.397-.994.265-.272.593-.409.982-.409s.725.137 1.005.41c.282.272.422.604.422.993 0 .39-.14.722-.422.994-.28.273-.615.41-1.005.41zm-1.099 13.376V12.398h2.268V23.74h-2.268zM132.508 23.74h-2.244V12.398h1.567l.397 1.496a8.144 8.144 0 012.059-1.344c.763-.351 1.52-.527 2.268-.527 1.231 0 2.147.383 2.747 1.146.6.764.901 1.793.901 3.087v7.484h-2.245v-7.273c0-.826-.137-1.454-.41-1.883-.272-.428-.76-.643-1.461-.643-.53 0-1.108.168-1.731.503-.623.335-1.239.768-1.848 1.298v7.998zM17.5 35C7.85 35 0 27.15 0 17.5S7.85 0 17.5 0 35 7.85 35 17.5 27.15 35 17.5 35zm8.473-21.16a2.28 2.28 0 01-4.112 1.36H2.9c-.118.75-.179 1.518-.179 2.3 0 8.15 6.63 14.78 14.78 14.78 5.968 0 11.121-3.556 13.452-8.66H13.495a2.28 2.28 0 11-.077-2.719h18.467c.259-1.093.395-2.23.395-3.4 0-8.15-6.63-14.78-14.78-14.78-6.388 0-11.842 4.073-13.902 9.758h18.263a2.28 2.28 0 014.112 1.36z"
            fill="#A6AAB4"
          />
        </svg>
      </div>
      <div
        class="MuiBox-root MuiBox-root"
      >
        <div
          class="MuiLinearProgress-root MuiLinearProgress-colorPrimary makeStyles-progressBar MuiLinearProgress-indeterminate"
          role="progressbar"
        >
          <div
            class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
          />
          <div
            class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
